sorl_thumbnail_serializer/fields.py

```diff
@@ -82,12 +84,14 @@ def to_representation(self, value):
 
         image = get_thumbnail(value, self.geometry_string, **self.options)
 
-        try:
-            request = self.context.get('request', None)
-            return request.build_absolute_uri(image.url)
-        except:
-            try:
-                return super(HyperlinkedSorlImageField, self).to_representation(image.url)
-            except AttributeError:  # NOQA
-                return super(HyperlinkedSorlImageField, self).to_native(image.url)  # NOQA
+        request = self.context.get('request', None)
+        assert request is not None, (
+            "`%s` requires the request in the serializer"
+            " context. Add `context={'request': request}` when instantiating "
+            "the serializer." % self.__class__.__name__
+        )
+
+        if self.uri_prefix:
+            return request.build_absolute_uri(self.uri_prefix + image.url)
```

Comment:
Is `uri_prefix` (and `settings.URI_PREFIX`) meant to for a path component or a complete URL, or both? I guess `build_absolute_uri()` would handle both.

I think I would prefer something like

```
from urllib.parse import urljoin
urljoin(self.uri_prefix, image.url)
```

or even `os.path.join(self.uri_prefix, image.url)` though.